### PR TITLE
[Security] sync translations and add a test for it

### PR DESCRIPTION
**Pull request by xabbuh**
| Q | A |
| --- | --- |
| Bug fix? | yes |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | #15955 |
| License | MIT |
| Doc PR |  |

### CODE REVIEW

**Comment by ogizanagi** [comment]
@xabbuh : Well didn't see your PR. I've closed mine (https://github.com/symfony/symfony/pull/16148) :smile:

File: src/Symfony/Component/Security/Tests/TranslationSyncStatusTest.php [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    private function getLegacyTranslationsDirectory()
+    {
+        return __DIR__.'/../Resources/translations';
+    }
+
+    private function getCoreTranslationsDirectory()
+    {
+        $securityClass = new \ReflectionClass('Symfony\Component\Security\Core\Security');
+
+        return dirname($securityClass->getFileName()).'/Resources/translations';
```
Comment (ogizanagi) [inline src/Symfony/Component/Security/Tests/TranslationSyncStatusTest.php]: When testing it from the Security component, I don't think there is the necessity to find the `translation` dir like this.
It'll always be into `__DIR__.'/../Core/Resources/translations'`, right ?

File: src/Symfony/Component/Security/Tests/TranslationSyncStatusTest.php [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+    public function testFileContentsAreEqual()
+    {
+        $finder = new Finder();
+        $files = $finder->in($this->getLegacyTranslationsDirectory())->files();
+
+        foreach ($files as $file) {
+            $coreFile = $this->getCoreTranslationsDirectory().'/'.basename($file);
+
+            $this->assertEquals(file_get_contents($file), file_get_contents($coreFile), $file.' and '.$coreFile.' have equal content.');
```
Comment (ogizanagi) [inline src/Symfony/Component/Security/Tests/TranslationSyncStatusTest.php]: Can use `assertFileEquals`

File: src/Symfony/Component/Security/Tests/TranslationSyncStatusTest.php [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+     */
+    public function testTranslationFileIsNotMissingInCore($dir1, $dir2)
+    {
+        $finder = new Finder();
+        $files = $finder->in($dir1)->files();
+
+        foreach ($files as $file) {
+            $filename = basename($file);
+
+            $this->assertTrue(file_exists($dir2.'/'.$filename), 'Missing file '.$filename.' in directory '.$dir2);
```
Comment (ogizanagi) [inline src/Symfony/Component/Security/Tests/TranslationSyncStatusTest.php]: `assertFileExists` ?

File: src/Symfony/Component/Security/Tests/TranslationSyncStatusTest.php (hunk 1 above)
Comment (stof) [inline src/Symfony/Component/Security/Tests/TranslationSyncStatusTest.php]: agreed

**Comment by xabbuh** [comment]
I addressed the comments.

**Comment by stof** [comment]
:+1:

**Comment by fabpot** [comment]
Thank you @xabbuh.

**Pull request merged**